.github/workflows/unit-test.yaml

```diff
@@ -0,0 +1,28 @@
+name: Unit Tests 
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  lint:
```

Comment:
This should be updated to match the job name. Also, can we just coalesce this and `test-microk8s.yaml` into `ci.yaml` or similar? See here for where I went with `ci.yaml` and `cd.yaml`:

https://github.com/canonical/spark-operator/tree/master/.github/workflows

I'm not tied to that though, this would be a good opportunity to do some bikeshedding.

Reply:
I like unifying things, although I do think functional tests should be in a separate workflow. I like the idea of 3 workflows, linting & unit-tests, functional tests and publishing.

Regarding naming, I'm not entirely sure what we should use, I do think using acronyms like CI and CD sometimes ends up being a little confusing.